src/core/Database.h

```diff
@@ -63,6 +63,15 @@ class Database : public ModifiableObject
     };
     static const quint32 CompressionAlgorithmMax = CompressionGZip;
 
+    enum SaveFlag
+    {
+        Atomic = 0x0, // Saves are transactional and atomic
+        NonAtomic = 0x1, // Save operation may be interrupted
+        Backup = 0x2, // Make copy of original database before saving
+        DirectWrite = 0x4, // Directly write to the file (dangerous)
+    };
```

Comment:
`Atomic`, `NonAtomic` and `DirectWrite` are mutually exclusive, yet you could set all three at the same time. You should remove `NonAtomic` and then use combinations of the two remaining flags:

- `Atomic`: atomic saving
- `DirectWrite`: very unsafe non-atomic saving
- `Atomic | DirectWrite`: unsafe temporary file

Perhaps there are better names for the flags to make the behaviour more obvious.

Reply:
True, the only logical combination is with Backup. We could convert the enum to straight integer and leave a boolean for backup in the function call. I was trying to eliminate random boolean.

Reply:
The alternative would be a fixed enum with all combinations, but that would be six options just because we have backup for everything.

Reply:
I fixed this by changing the enum to an integer and adding a backup boolean to the function call.